tracer/test/Datadog.Trace.TestHelpers/MockTelemetryAgent.cs

```diff
@@ -118,7 +118,7 @@ public TelemetryData WaitForLatestTelemetry(
                     }
                 }
 
-                Thread.Sleep(sleepTime);
+                await Task.Delay(sleepTime).ConfigureAwait(false);
```

Comment:
I think we should be safe
```suggestion
                await Task.Delay(sleepTime);
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/7145/commits/140b6ac9b377428380005da150f82e712b168cc1